drivers/gpio/gpio_cc32xx.c

```diff
@@ -46,42 +64,42 @@ struct gpio_cc32xx_data {
 #define DEV_DATA(dev) \
 	((struct gpio_cc32xx_data *)(dev)->driver_data)
 
+static int gpio_cc32xx_port_set_bits_raw(struct device *port, u32_t mask);
+static int gpio_cc32xx_port_clear_bits_raw(struct device *port, u32_t mask);
+
 static inline int gpio_cc32xx_config(struct device *port,
 				   int access_op, u32_t pin, int flags)
 {
 	const struct gpio_cc32xx_config *gpio_config = DEV_CFG(port);
 	unsigned long port_base = gpio_config->port_base;
-	unsigned long int_type;
 
-	/*
-	 * See pinmux_initialize(): which leverages TI's recommended
-	 * method of using the PinMux utility for most pin configuration.
-	 */
+	if (((flags & GPIO_INPUT) != 0) && ((flags & GPIO_OUTPUT) != 0)) {
+		return -ENOTSUP;
+	}
 
-	if (access_op == GPIO_ACCESS_BY_PIN) {
-		/* Just handle runtime interrupt type config here: */
-		if (flags & GPIO_INT) {
-			if (flags & GPIO_INT_EDGE) {
-				if (flags & GPIO_INT_ACTIVE_HIGH) {
-					int_type = GPIO_RISING_EDGE;
-				} else if (flags & GPIO_INT_DOUBLE_EDGE) {
-					int_type = GPIO_BOTH_EDGES;
-				} else {
-					int_type = GPIO_FALLING_EDGE;
-				}
-			} else { /* GPIO_INT_LEVEL */
-				if (flags & GPIO_INT_ACTIVE_HIGH) {
-					int_type = GPIO_HIGH_LEVEL;
-				} else {
-					int_type = GPIO_LOW_LEVEL;
-				}
-			}
-			MAP_GPIOIntTypeSet(port_base, (1 << pin), int_type);
-			MAP_GPIOIntClear(port_base, (1 << pin));
-			MAP_GPIOIntEnable(port_base, (1 << pin));
+	if ((flags & (GPIO_INPUT | GPIO_OUTPUT)) == 0) {
+		return -ENOTSUP;
+	}
+
+	if ((flags & (GPIO_PULL_UP | GPIO_PULL_DOWN)) != 0) {
+		return -ENOTSUP;
+	}
+
+	if (access_op != GPIO_ACCESS_BY_PIN) {
+		return -ENOTSUP;
+	}
+
+	MAP_PinTypeGPIO(pinTable[gpio_config->port_num * 8 + pin],
+		PIN_MODE_0, false);
+	if (flags & GPIO_OUTPUT) {
+		MAP_GPIODirModeSet(port_base, (1 << pin), GPIO_DIR_MODE_OUT);
```

Comment:
We should enable the output only after we set its initial value.

Reply:
Unfortunately setting the output mode after setting its initial value leads to occasional failures in the gpio_basic_api test. The initial value doesn't seem to be correctly set on all runs. Both the TRM and the code in TIRTOS are showing to setting the mode first. So I am leaving this unchanged.

Reply:
OK, if the hardware doesn't support it there is nothing we can do.